v2/devices/hero2lte.yml

```diff
@@ -0,0 +1,106 @@
+    button: true
+  download:
+    title: "Reboot to Download Mode"
+    description: "Press and hold Volume Down + Home + Power, until you see the Warning about Custom OS installations, press Volume Up to Continue, then you may have to unplug and plug the USB cord back in."
+    image: "phone_power_down"
+    button: true
+  hero2lte_check:
+    title: "Unlock Bootloader"
+    description: "You need an unlocked bootloader. If your bootloader was never unlocked, please refer to the first two oem unlock steps linked here."
+    link: "https://forum.xda-developers.com/t/recovery-exynos-official-twrp-for-galaxy-s7-edge-hero2lte.3334084/"
+unlock: []
+handlers:
+  bootloader_locked:
+    actions:
+      - fastboot:oem_unlock:
+operating_systems:
+  - name: "Ubuntu Touch"
+    compatible_installer: ">=0.9.2-beta"
+    options:
+      - var: "channel"
+        name: "Channel"
+        tooltip: "The release channel"
+        link: "https://docs.ubports.com/en/latest/about/process/release-schedule.html"
+        type: "select"
+        remote_values:
+          systemimage:channels:
+      - var: "wipe"
+        name: "Wipe Userdata"
+        tooltip: "Wipe personal data (required for new installs)"
+        type: "checkbox"
+    prerequisites:
+      - "hero2lte_check"
+    steps:
+      - actions:
+          - core:download:
+              group: "firmware"
+              files:
+                - url: "https://github.com/Kethen/samsung-exynos8890/releases/download/2022-07-23/recovery_hero2lte.img"
+                  checksum:
+                    sum: "c09e3fed9f8d48be83123c9edfba70a81f7399791833008179fe006eb5715d47"
+                    algorithm: "sha256"
+                - url: "https://github.com/Kethen/samsung-exynos8890/releases/download/2022-07-23/boot-reboot-recovery-hero2lte.img"
+                  checksum:
+                    sum: "d04b0f15a52c667b723663ca110e1f523d52a4e97823670a378663a61808eb7e"
+                    algorithm: "sha256"
+      - actions:
+          - core:user_action:
+              action: "download"
+      - actions:
+          - heimdall:flash:
+              partitions:
+                - partition: "BOOT"
+                  file: "boot-reboot-recovery-hero2lte.img"
+                  group: "firmware"
+                - partition: "RECOVERY"
+                  file: "recovery_hero2lte.img"
+                  group: "firmware"
+      - actions:
+          - adb:reboot:
+              to_state: "recovery"
+        fallback:
+          - core:user_action:
+              action: "recovery"
+      - actions:
+          - adb:shell:
+              args: ["mkfs.ext4", "/dev/block/platform/155a0000.ufs/by-name/USERDATA"]
+        condition:
+          var: "wipe"
+          value: true
+      - actions:
+          - adb:shell:
+              args: ["mount", "/data"]
+        condition:
+          var: "wipe"
+          value: true
+      - actions:
+          - adb:shell:
+              args: ["mkdir", "/data/cache"]
```

Comment:
The system should be able to detect and create that itself instead of relying on the installer, shouldn't it?

Reply:
That is only for after wiping, guarded by the "wipe" var

Reply:
Yes, that's what i mean :)

Init should check for its existence and recreate it, if it does not exist, instead of the installer.

Reply:
Hmm.. I'm probably missing something then on the recovery

I didn't find a way to write the fstab so that `ensure_path_mounted` would preform mounting /data and creating /data/cache for me with just /cache mount, so these modifications were made to redirect the cache mount with a symbolic link, and create /data/cache if needed on recovery start

https://github.com/Kethen/halium_bootable_recovery/commit/51eb0798308af2a8db61d836bff51e7f2a7b2ef4

/cache mount was disabled in fstab meanwhile

https://github.com/Kethen/android_device_samsung_universal8890-common/blob/lineage-18.1-ut-recovery/ramdisk/etc/fstab.samsungexynos8890

That's why in the installer config, after the wipe with mkfs.ext4, the next step is to re-create /data/cache before `systemimage:install:`